src/gtk/webview_webkit2.cpp

```diff
@@ -1654,6 +1662,63 @@ void wxWebViewWebKit::RemoveAllUserScripts()
     webkit_user_content_manager_remove_all_scripts(ucm);
 }
 
+
+#ifdef wxHAVE_WEBKIT_WEBSITE_DATA_MANAGER_CLEAR
+static void
+wxgtk_webview_webkit_clear_data_ready(GObject *,
+                                      GAsyncResult *result,
+                                      void *user_data)
+{
+    wxWebViewWebKit* webKitCtrl = static_cast<wxWebViewWebKit*>(user_data);
+    WebKitWebsiteDataManager* manager = static_cast<wxWebViewConfigurationImplWebKit*>(webKitCtrl->m_config.GetImpl())->GetWebsiteDataManager();
+
+    gboolean success = webkit_website_data_manager_clear_finish(manager, result, nullptr);
+    wxWebViewEvent event(wxEVT_WEBVIEW_BROWSING_DATA_CLEARED,
+                         webKitCtrl->GetId(),
+                         webKitCtrl->GetCurrentURL(),
+                         "");
+    event.SetEventObject(webKitCtrl);
+    event.SetInt((success) ? 1 : 0);
+    webKitCtrl->HandleWindowEvent(event);
+}
+#endif // wxHAVE_WEBKIT_WEBSITE_DATA_MANAGER_CLEAR
+
+bool wxWebViewWebKit::ClearBrowsingData(int types, wxDateTime since)
+{
+#ifdef wxHAVE_WEBKIT_WEBSITE_DATA_MANAGER_CLEAR
+    if (wx_check_webkit_version(2, 16, 0))
+    {
+        WebKitWebsiteDataManager* manager = static_cast<wxWebViewConfigurationImplWebKit*>(m_config.GetImpl())->GetWebsiteDataManager();
+
+        if (types & wxWEBVIEW_BROWSING_DATA_OTHER)
+            return false;
```

Comment:
Why not map it to all the elements of `WebKitWebsiteDataTypes` not otherwise used below?